### PR TITLE
Revert "Make sure to replace the used refresh token with a fresh one"

### PR DESCRIPTION
**Pull request by Netlob**
I forgot that the refresh token isnt updated on refresh. The Spotistats API makes use of the authorization code flow without pkce, and explained here: https://developer.spotify.com/documentation/general/guides/authorization-guide/#4-requesting-a-refreshed-access-token-spotify-returns-a-new-access-token-to-your-app the refresh accesstoken request does _not_ return a new refresh token.

tldr; refresh token doesn't change and can be used infinitely

ps. `refresh_token` doesn't even exist on the `RefreshAccessTokenResponse` interface:
```ts
interface RefreshAccessTokenResponse {
    access_token: string;
    expires_in: number;
    scope: string;
    token_type: string;
}
```

### CODE REVIEW

**Comment by Netlob** [comment]
reverted #2

**Comment by rschildmeijer** [comment]
Hey again Netlob,

You _are_ actually using PKCE, just verified using the Android app (v. 1.0.7). The authorize url used for Android looked like this:
https://accounts.spotify.com/en/authorize?redirect_uri=spotistats:%2Foauthredirect&client_id=7c4d72c7d50a47fcb2e7fc6c29120e4b&
response_type=code&
state=ZuYE-7PQoAfLk0ZBjCp_ag&
scope=user-read-playback-state%20user-read-currently-playing%20user-read-email%20user-read-private%20playlist-read-collaborative%20playlist-modify-public%20playlist-read-private%20playlist-modify-private%20user-library-modify%20user-library-read%20user-top-read%20user-read-playback-position%20user-read-recently-played%20user-follow-read%20user-follow-modify&
**code_challenge**=Lcr4HaNLNgNJ-OL_BrncVwPp7o8nI550z9CGh2lioso&
**code_challenge_method**=S256

The last two query params indicates that you are using PKCE. Also, on the server when exchanging the authorization code for access token and refresh token you are following the PKCE specification: see [](https://github.com/Netlob/spotistats-api/blob/main/src/routers/auth.ts#L59)

So, you **will** get back a new refresh token when requesting a access token which you need to persist in your database.

**Comment by Netlob** [comment]
Hi Roger,

First of all: thanks for your contribution to this project! 

You are right about the app using the PKCE flow, but currently, the app has nothing to do with the server. The login in the app is pure client-side PKCE as we indeed see by the URL. The login in this repo _was_ not using PKCE since it has access to the client secret.

But, in order to allow for more features, the next version of the app will log in through this server. The new flow will then be:
1. app opens a login popup
2. user authorizes
3. popup will close and give 2 variables: `authorizationCode` and `codeVerifier`
4. these are sent to this API  and exchanged for a token set
5. the refresh token from this set _does_ need to get refreshed too

So basically you made a 100% correct PR for a future commit 🎉 

ps. sorry for the messy code :p
